shared/reducers/fs.tsx

```diff
@@ -356,38 +390,51 @@ export default Container.makeReducer<FsGen.Actions, Types.State>(initialState, {
     draftState.tlfUpdates = action.payload.tlfUpdates
   },
   [FsGen.dismissFsError]: (draftState, action) => {
-    draftState.errors = draftState.errors.remove(action.payload.key)
+    if (draftState.errors.has(action.payload.key)) {
+      const errors = new Map([...draftState.errors])
+      errors.delete(action.payload.key)
+      draftState.errors = errors
+    }
   },
   [FsGen.showMoveOrCopy]: (draftState, action) => {
-    draftState.destinationPicker = draftState.destinationPicker
-      .update('source', source =>
-        source.type === Types.DestinationPickerSource.MoveOrCopy ? source : Constants.makeMoveOrCopySource()
-      )
-      .set('destinationParentPath', I.List([action.payload.initialDestinationParentPath]))
+    draftState.destinationPicker.source =
+      draftState.destinationPicker.source.type === Types.DestinationPickerSource.MoveOrCopy
+        ? draftState.destinationPicker.source
+        : ({
+            path: Constants.defaultPath,
+            type: Types.DestinationPickerSource.MoveOrCopy,
+          } as const)
+
+    draftState.destinationPicker.destinationParentPath = [action.payload.initialDestinationParentPath]
   },
   [FsGen.setMoveOrCopySource]: (draftState, action) => {
-    draftState.destinationPicker = draftState.destinationPicker.set(
-      'source',
-      Constants.makeMoveOrCopySource({path: action.payload.path})
-    )
+    draftState.destinationPicker.source = {
+      path: action.payload.path,
+      type: Types.DestinationPickerSource.MoveOrCopy,
+    }
   },
   [FsGen.setDestinationPickerParentPath]: (draftState, action) => {
-    draftState.destinationPicker = draftState.destinationPicker.update('destinationParentPath', list =>
-      list.set(action.payload.index, action.payload.path)
-    )
+    if (draftState.destinationPicker.destinationParentPath[action.payload.index] !== action.payload.path) {
+      const destinationParentPath = [...draftState.destinationPicker.destinationParentPath]
+      destinationParentPath[action.payload.index] = action.payload.path
+      draftState.destinationPicker.destinationParentPath = destinationParentPath
```

Comment:
you don't need to copy to mutate arrays in immer
`draftState.destinationPicker.destinationParentPath[action.payload.index] = action.payload.path`